elastic/elastic-agent
# Write `container-paths.yml` into the `STATE_PATH`

## What does this PR do?



Adjusts the `container` subcommand to write the `container-paths.yml` into the `STATE_PATH` on startup.

## Why is it important?



When running the Elastic Agent container with `--read-only` it is not possible for the container to write the `container-paths.yml` files into the root filesystem of the container.

## Checklist




## How to test this PR locally



```
$ PLATFORMS="linux/arm64" PACKAGES="docker" SNAPSHOT=true EXTERNAL=true mage package
...
$ docker run --rm --mount type=tmpfs,destination=/state -e STATE_PATH=/state --read-only docker.elastic.co/beats/elastic-agent:8.15.0-SNAPSHOT
```

In another shell with the container running verify that the `status` command works.

```
$ docker exec -it ${container_id} bash
elastic-agent@7d7447b8f608:~$ elastic-agent status
┌─ fleet
│  └─ status: (STOPPED) Not enrolled into Fleet
└─ elastic-agent
   └─ status: (HEALTHY) Running
```
